game/src/main/java/org/apollo/game/model/area/collision/CollisionManager.java

```diff
@@ -25,20 +27,14 @@
 public final class CollisionManager {
 
 	/**
-	 * A comparator that sorts {@link Position}s by their X coordinate, then Y, then height.
+	 * A {@code HashMultimap} of region coordinates mapped to positions where the tile is completely blocked.
 	 */
-	private static final Comparator<Position> POSITION_COMPARATOR =
-		Comparator.comparingInt(Position::getX).thenComparingInt(Position::getY).thenComparingInt(Position::getHeight);
+	private final Multimap<RegionCoordinates, Position> blocked = HashMultimap.create();
```

Comment:
Do we need the Map semantics here?  I think this can also be a HashSet.

Reply:
I think this is faster because the other option would be looping over every tile in the region and doing a `HashSet#contains` before applying the tile as blocked. With the map we know which tiles are blocked already.

Reply:
Er, right.  I missed the fact that we're actually iterating over these positions.